src/main/java/competition/subsystems/drive/DriveSubsystem.java

```diff
@@ -32,17 +33,32 @@ public DriveSubsystem(XCANMotorController.XCANMotorControllerFactory motorContro
         this.frontRight = motorControllerFactory
                 .create(new CANMotorControllerInfo("FrontRight", 2), this.getPrefix(), "FrontRight");
 
+        this.motorMultiplier = 1.0; // init with no precision mode
+
         pf.setPrefix(this);
         dp = pf.createPersistentProperty("DriveSubsystem", 1.5);
     }
 
-    public void tankDrive(double leftPower, double rightPower) {
+    public void tankDrive(double leftInput, double rightInput) {
         // You'll need to take these power values and assign them to all of the motors.
         // As an example, here is some code that has the frontLeft motor to spin
         // according to the value of leftPower:
-        frontLeft.setPower(leftPower);
-        // TODO: Add code to set the right motors to the rightPower value.
+        frontLeft.setPower(leftInput * motorMultiplier);
+        frontRight.setPower(rightInput * motorMultiplier);
+    }
+
+    public double getMotorMultiplier() {
+        return motorMultiplier;
+    }
 
+    public void setMotorMultiplier(double multiplier) {
+        if (multiplier > 1.0) {
```

Comment:
Another nice concise way of doing this is with `MathUtil.clamp(multiplier, -1, 1)`